apps/docs/content/guides/realtime/authorization.mdx

```diff
@@ -0,0 +1,288 @@
+```
+
+### Broadcast
+
+The `extension` field on the `realtime.messages` table records the message type. For Broadcast messages, the value of `realtime.messages.extension` is `broadcast`. You can check for this in your RLS policies.
+
+#### Allow a user to join (and read) a Broadcast topic
+
+To join a Broadcast Channel, a user must have at least one read or write permission on the Channel topic.
+
+Here, we allow reads (`select`s) for users who are linked to the requested topic within the relationship table `public.room_users`:
+
+```sql
+create policy "authenticated users can join topic"
+on "realtime"."messages"
+for select
+to authenticated
+using (
+exists (
+    select
+      ru.user_id
+    from
+      rooms_users ru
+    where
+      ru.user_id = (select auth.uid())
+      and ru.topic = (select realtime.topic())
+      and realtime.messages.extension in ['broadcast']
+  )
+);
+```
+
+Then, to join a topic with RLS enabled, instantiate the Channel with the `private` option set to `true`.
+
+```typescript
+import { createClient } from 'npm:@supabase/supabase-js@2.38.5'
+const url = 'https://<project_ref>.supabase.com'
+const apikey = '<api_key>'
+
+const client = createClient(url, apikey)
+
+const channel = client.channel('room_1', {
+  config: { private: true },
+})
+
+channel
+  .on('broadcast', { event: 'test' }, (payload) => console.log(payload))
+  .subscribe((status: string, err: any) => {
+    if (status === 'SUBSCRIBED') {
+      console.log('Connected!')
+    } else {
+      console.error(err)
+    }
+  })
+```
+
+#### Allow a user to send a Broadcast message
+
+To authorize sending Broadcast messages, create a policy for `insert` where the value of `realtime.messages.extension` is `broadcast`.
+
+Here, we allow writes (sends) for users who are linked to the requested topic within the relationship table `public.room_users`:
+
+```sql
+create policy "authenticated users can send broadcast messages in a room"
+on "realtime"."messages"
+for insert
+to authenticated
+with check (
+  exists (
+    select
+      ru.user_id
+    from
+      rooms_users ru
+    where
+      ru.user_id = (select auth.uid())
+      and ru.topic = (select realtime.topic())
+      and realtime.messages.extension in ['broadcast']
+  )
+);
+```
+
+### Presence
+
+The `extension` field on the `realtime.messages` table records the message type. For Presence messages, the value of `realtime.messages.extension` is `presence`. You can check for this in your RLS policies.
+
+#### Allow users to listen to Presence messages on a Channel
+
+Create a policy for `select` on `realtime.messages` where `realtime.messages.extension` is `presence`.
+
+```sql
+create policy "authenticated users can listen to presence changes in a room"
+on "realtime"."messages"
+for select
+to authenticated
+using (
+  exists (
+    select
+      ru.user_id
+    from
+      rooms_users ru
+    where
+      ru.user_id = (select auth.uid())
+      and ru.topic = (select realtime.channel_name())
```

Comment:
Is `realtime.channel_name()` another helper function that we need to document (alongside `realtime.topic`)? What is the difference between these two (since they both seem to relate to the `room_users.topic` column?

Reply:
realtime.channel_name() no longer exists and was removed, realtime.topic() took it's place

Reply:
Fixed.